app/src/routes.tsx

```diff
@@ -0,0 +1,214 @@
+            { title: `${lang.USER_MANAGEMENT}`, key: 'user', url: `${URL_USER}` },
+            { title: `${lang.USER_GROUPS}`, key: 'user-groups', url: `${URL_USER_GROUPS}` },
+            { title: `${lang.USER_ROLES}`, key: 'user-roles', url: `${URL_USER_ROLES}` },
+          ],
+        },
+        {
+          title: 'Locations',
+          key: 'location',
+          enabled:
+            ENABLE_LOCATIONS &&
+            roles &&
+            activeRoles.LOCATIONS &&
+            isAuthorized(roles, activeRoles.LOCATIONS.split(',')),
+          children: [
+            { title: `${lang.LOCATION_UNIT}`, url: `${URL_LOCATION_UNIT}`, key: 'location-unit' },
+            {
+              title: `${lang.LOCATION_UNIT_GROUP}`,
+              url: `${URL_LOCATION_UNIT_GROUP}`,
+              key: 'location-group',
+            },
+          ],
+        },
+        {
+          title: `${lang.PRODUCT_CATALOGUE}`,
+          key: 'product-catalogue',
+          enabled:
+            ENABLE_PRODUCT_CATALOGUE &&
+            roles &&
+            activeRoles.PRODUCT_CATALOGUE &&
+            isAuthorized(roles, activeRoles.PRODUCT_CATALOGUE.split(',')),
+          url: `${CATALOGUE_LIST_VIEW_URL}`,
+        },
+        {
+          title: `${lang.TEAMS}`,
+          key: 'teams',
+          enabled:
+            ENABLE_TEAMS &&
+            roles &&
+            activeRoles.TEAMS &&
+            isAuthorized(roles, activeRoles.TEAMS.split(',')),
+          children: [
+            { title: `${lang.TEAMS}`, url: `${URL_TEAMS}`, key: 'teams-list' },
+            {
+              title: `${lang.TEAM_ASSIGNMENT}`,
+              url: `${URL_TEAM_ASSIGNMENT}`,
+              key: 'team-assignment',
+            },
+          ],
+        },
+        {
+          title: `${lang.FORM_CONFIGURATION}`,
+          key: 'form-config',
+          enabled:
+            ENABLE_FORM_CONFIGURATION &&
+            roles &&
+            activeRoles.FORM_CONFIGURATION &&
+            isAuthorized(roles, activeRoles.FORM_CONFIGURATION.split(',')),
+          children: [
+            {
+              title: `${lang.MANIFEST_RELEASES}`,
+              key: 'form-config-releases',
+              url: `${URL_MANIFEST_RELEASE_LIST}`,
+            },
+            {
+              title: `${lang.DRAFT_FILES}`,
+              key: 'form-config-draft',
+              url: `${URL_DRAFT_FILE_LIST}`,
+            },
+            {
+              title: `${lang.JSON_VALIDATORS}`,
+              key: 'form-config-validators',
+              url: `${URL_JSON_VALIDATOR_LIST}`,
+            },
+          ],
+        },
+      ],
+    },
+  ];
+
+  function filterfalsy(route: Route[]): Route[] {
+    return route
+      .filter(
+        (e) => !e.hasOwnProperty('enabled') || (e.hasOwnProperty('enabled') && e.enabled === true)
+      )
+      .map((e) => {
+        if (e.children) return { ...e, children: filterfalsy(e.children) };
+        else return e;
+      });
+  }
```

Comment:
lets avoid nested functions where possible, obviously makes unit testing impossible.

Reply:
sure let me move this outside